k8s/src/main/scala/io/buoyant/k8s/v1.scala

```diff
@@ -72,12 +72,43 @@ package object v1 {
     def toWatch(e: Service) = ServiceModified(e)
   }
 
+  @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
```

Comment:
I'd include the changes in this file with your next branch, instead of this one.